Implemented api.enableCue(), api.disableCue() and api.extendOptions().

metincansiper opened the pull request:
Implemented api.enableCue(), api.disableCue() to enable/disble the visual cue.

Also in Readme updated description of `cy.expandCollapse(options)` as "To initialize the extension with given options.". The old description was "To initialize/set options whenever you want.". However, it was looking like that method would not be working fine just to update/extend the user options (it was only good for initializing the extension). There was already existing `api.setOption(name, val)` and `api.setOptions(opts)` methods to update the options so I added `api.extendOptions(opts)` as well that extends the existing options with the given ones.

hasanbalci commented:
@metincansiper A point that I noticed: api.extendOptions(opts) automatically disables cue when the opts doesn't include cueEnabled.  

Also please add this new functionality to the readme.

metincansiper commented:
@hasanbalci I fixed that bug and added the missing method description to README. Then I merged the latest unstable to the feature branch to get rid of conflicts and make the code compatible with latest unstable branch code base.

hasanbalci commented:
@metincansiper I think when we change options by using api.extendOptions(opts), the changes aren't applied to operations done with visual cue. For example, if we use `api.extendOptions({fisheye: false});` and expand a node by using api.expand(), fisheye isn't applied which is correct. But if I expand it by using visual cue, it's applied which shouldn't be the case.

hasanbalci commented on src/index.js:
```diff
@@ -175,11 +200,27 @@
+      api.disableCue = function() {
+        var options = getScratch(cy, 'options');
+        if (options.cueEnabled) {
+          cueUtilities('unbind', cy, api, $);
```
'$' sign here and below must be removed, since jQuery dependency is removed from this extension.

metincansiper commented:
@hasanbalci actually that bug must not be directly related to this PR but I suspect it maybe related to something missing while merging the latest unstable into this branch (it was more than a simple merge actually because of the some changes in the codebase of unstable meanwhile). $ sign problem was also related. Both must be fixed by my last commit to this PR.

ugurdogrusoz commented:
I am not sure if this is related but now when I am on a visual cue to collapse, which happens to be on an edge as well, event is considered by only the edge, and collapse will not take place. For instance clicking on the visual cue will only select the edge compartment will not be collapsed (both should happen):
![image](https://user-images.githubusercontent.com/3874988/62765722-66dc7f80-ba99-11e9-87c4-063818aec6e8.png)

ugurdogrusoz commented:
According to @hasanbalci, this is due to [this change](https://github.com/iVis-at-Bilkent/cytoscape.js-edge-editing/commit/62695f276a79d1c729260259fd0bae294bfe0987) by @nasimsaleh.
